modules/promotion/src/Element/CouponRedemptionForm.php

```diff
@@ -81,7 +88,26 @@ public static function processForm(array $element, FormStateInterface $form_stat
       throw new \InvalidArgumentException('The commerce_coupon_redemption #order_id must be a valid order ID.');
     }
 
-    $has_coupons = !$order->get('coupons')->isEmpty();
+    /** @var \Drupal\commerce_promotion\Entity\CouponInterface[] $coupons */
+    $coupons = $order->get('coupons')->referencedEntities();
+    $has_coupons = !empty($coupons);
+
+    // Determine if 'add coupon' buttons should be displayed. It should be
+    // on display_actions mode, if there is no coupons, or if at least one of
+    // the coupons is compatible with others.
+    $display_actions = !$has_coupons;
+
+    if (!$element['#single_coupon_mode']) {
+      /** @var \Drupal\commerce_promotion\Entity\CouponInterface $coupon */
+      foreach ((array) $coupons as $coupon) {
+        if ($coupon->getPromotion()->getCompatibility() == PromotionInterface::COMPATIBLE_ANY) {
```

Comment:
We should remove this. Not this element's job to respect compatibility. That's on redemption validation.